src/parser.js

```diff
@@ -1148,6 +1156,19 @@ function mapSwitchStatement(node, meta) {
 }
 
 function mapForStatement(node, meta) {
+  let blockStatement = mapBlockStatement(node.body, meta);
+
+  if (node.guard) {
+    const guardBase = node.guard.expression || node.guard;
+
+    blockStatement = b.blockStatement([
+      b.ifStatement(
+        mapOp(guardBase, meta),
```

Comment:
Will it always be correct to `mapOp` on the guard? Presumably you could have a conditional like `foo for foo in foos when foo === 'bar' && foo.toUpperCase() === 'BAR'`